### DISCUSSION TITLE
Multiple models types (more than one dropdown menu)

### DISCUSSION THREAD

**Discussion by Rodien** [openstore-ecommerce/OpenStore#133] [General]
Case: 

Shop owner sells shoes and he wants a product to a have a model for the colors and a model for the shoe sizes. 
This way the shop owner has control over the stock of colors and sizes. 

For example, the shop owner can now mark shoe model size 10 en model of color red as out of stock. 

Is there a way to achieve this in open store?

**Comment by DNNMonster** [comment]
OpenStore tracks stock at the model level.

If you're using models(and not options) then you should be able to track the stock of each model in the manner that you are describing.

I'm not 100% sure but it sounds like you may be using options...  If that's the case then you would bump into difficulties.

There would need to be some changes to add support for option stock tracking in the core.  

Another viable option would be to implement a custom OrderCompleteEvent handler that would allow for you to track stock per option.

**Reply by Rodien** [reply]
Thank you for reacting so fast😁

I have tried to add more than one model for a product but what then happens is that a customer can choose just one of the model options. They can either choose e.g. shoe size 10 or shoe color red but not both. So, it's not possible in Open Store.

We would have to introduce a new feature something like model grouping. Where on product can have more than one model group. A model group for colors, sizes, men, woman .ect 

With model grouping you could do things live color blue has 15% discount.

**Reply by leedavi** [reply]
Like DNNmonster says.  This can be done with a bespoke plugin to calculate stock on options, but it'll be a large amount of code to get this to work.  Changing the core code for things like this is avoided, and would probably be more work than a plugin.  The core code is stable and we have decided changes to the core will be restricted to essential requirements/fixes.

you can do what you want by create multiple models for each....

1 = Shoe 10 blue
2 = Shoe 10 red
3 = Shoe 10 green
4 = Shoe 11 blue
5 = Shoe 11 red

Think of models as an "object", the color is not an option but a product,  This way stock will work.  And if you think about it, having stock on a object (Model of Shoe) makes sense.  You can style the model list to look like options, hide the modeldropdown and use razor to list the shoes and options of the product, the when one is clicked you can change the modeldropdown to matchthe selection.

So, yes you can do this, with no code change to OpenStore.  But you'll need to alter the way the models are in the product and then alter the template to display what you need.

**Reply by leedavi** [reply]
BTW you have a sale price on each model, so you can do you discount on each model.
